### PR TITLE
feat: deprecate ClickHouse recordings ingestion

### PR DESCRIPTION
**Pull request by pauldambra**
We want to stop ingesting recordings into ClickHouse.

This has three components

1) doing this for PostHog cloud
2) doing this in tests/debug
3) doing this for self-hosted

As a first step we add an environment variable `SESSION_RECORDING_ALLOW_V1_INGESTION` which both the plugin server and django app read as config.

When it is true we:
* warn in the console
* start recordings ingestion capability 
* ingest to the `session_recordings_events` topic as well as the `session_recording_snapshot_items_events` topic

When it is false we:
* do not start the recordings ingestion capability
* drop events that would otherwise be ingested to the `session_recordings_events`

This would let us set the environment variable for cloud whenever we want as it wouldn't change behavior for us but would stop us writing to a topic we ignore

But would also give us a stepping stone to deprecating self hosted. 

This version warns
A future version is the final image that supports ClickHouse ingestion
And any self-hosted deployment that needed to continue that ingestion would have to pin to that docker image tag

That gives us a line in the sand after which we can tidy up our code

### CODE REVIEW

File: posthog/api/capture.py [hunk 1]
```diff
@@ -367,20 +371,16 @@ def get_event(request):
...
             replay_events, other_events = split_replay_events(events)
             processed_replay_events = replay_events
 
-            if len(replay_events) > 0:
-                # Legacy solution stays in place
+            if len(replay_events) > 0 and settings.SESSION_RECORDING_ALLOW_V1_INGESTION:
+                # Legacy solution stays in place to allow self-hosted systems a chance to upgrade
+                # or to pin their version so that they can continue to ingest recordings to ClickHouse
+                # while not receiving other fixes and upgrades
                 processed_replay_events = legacy_preprocess_session_recording_events_for_clickhouse(replay_events)
```
Comment (benjackwhite) [inline posthog/api/capture.py:382]: I think we need to do something about the line above as this change simply doesn't preprocess...

File: posthog/api/capture.py [hunk 2]
```diff
@@ -367,20 +371,16 @@ def get_event(request):
             # NOTE: Whilst we are testing this code we want to track exceptions but allow the events through if anything goes wrong
             capture_exception(e)
 
-        consumer_destination = "v2" if random() <= settings.REPLAY_EVENTS_NEW_CONSUMER_RATIO else "v1"
-
         try:
             replay_events, other_events = split_replay_events(events)
             processed_replay_events = replay_events
```
Comment (benjackwhite) [inline posthog/api/capture.py:376]: this is the line i mean

**Review by benjackwhite — CHANGES_REQUESTED** [review]
I think there is literally one line that is not correct. Otherwise looks good

**Comment by pauldambra** [comment]
gonna nuke this and pick it up in https://github.com/PostHog/posthog/pull/17572

**Pull request closed without merging**